.gitattributes

```diff
@@ -0,0 +1,25 @@
+# https://help.github.com/articles/dealing-with-line-endings/
+
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto
+
+# Explicitly declare text files you want to always be normalized and converted
+# to native line endings on checkout.
```

Comment:
No, no particular reason, I think it's been just an oversight. the default rule should already handle most cases as expected, the example file from the github documentation provided rules for C modules and I simply extended it with those for C++. I'll take the occasion to collect some information about the different file extensions used within the repository and verify if there are any other relevant cases that should be added.